src/io/perun.clj

```diff
@@ -344,13 +344,13 @@
 
 (def ^:private +atom-defaults+
   {:filename "atom.xml"
-   :filterer :content
+   :filterer :body
    :out-dir "public"})
 
 (deftask atom-feed
   "Generate Atom feed"
   [f filename    FILENAME    str  "generated Atom feed filename"
-   _ filterer    FILTER      code "predicate to use for selecting entries (default: `:content`)"
+   _ filterer    FILTER      code "predicate to use for selecting entries (default: `:body`)"
```

Comment:
fwiw, my interpretation was that `:content` was added to the metadata in `collection` so that collection pages would be present in sitemaps by default. I agree that something more fine-grained would be desirable. Seems like that should be a separate pull request though?

On Nov 18, 2016, 8:06 AM -0600, Martin Klepsch <notifications@github.com>, wrote:
>
> In src/io/perun.clj (https://github.com/hashobject/perun/pull/109#discussion_r88661956):
>
> > :out-dir "public"}) > > (deftask atom-feed > "Generate Atom feed" > [f filename FILENAME str "generated Atom feed filename" > - _ filterer FILTER code "predicate to use for selecting entries (default: `:content`)" > + _ filterer FILTER code "predicate to use for selecting entries (default: `:body`)"
> All files the render and collection tasks rendered had to have a :content key.
> That key was set by the markdown task usually but could have been set by other tasks in theory.
> I think setting :content to the rendered HTML inside the collection task (as in the line you linked (https://github.com/hashobject/perun/blob/master/src/io/perun.clj#L498)) does not make much sense but I might be missing something.
> Now with :body being the default filterer tasks after the render and collection tasks would see all the files rendered by those tasks. Previously they only respected files that had the :content key — in this case indicating they are original units of content. (With the exception of files rendered by the collection task. Not sure why :content is being set there.)
>
>
>
> It seems that the :content key was added to the files generated by the collection task here: 69a7149 (https://github.com/hashobject/perun/commit/69a7149108541bb4910fd8ef1a8ce7396d07c35b). I think overloading the meaning of the :content key like this can end up being very confusing so I'm thinking that maybe we should reconsider this one @podviaznikov (https://github.com/podviaznikov) (perhaps use :body or something similar as @bhagany (https://github.com/bhagany) does in this PR)
>
>
> In this regard it may also make sense to start using namespaced keywords like io.perun/markdown or whatever to indicate what can be expected to be the value of these keys.
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub (https://github.com/hashobject/perun/pull/109/files/fc71654711af0c716d5374eb045a3bdedaa7fef3#r88661956), or mute the thread (https://github.com/notifications/unsubscribe-auth/AAbIcpUif3NwAKNjYfWDo21DckWthHktks5q_bDvgaJpZM4K2Mwt).
>
>
>